server/client_manager.py

```diff
@@ -431,6 +489,24 @@ def change_music(self, song, cid, showname="", effects=0, loop=True):
                             "This URL is not allowed."
                         )
                         return
+                yt_parse = urlparse(song)
+
+                if "youtube" in yt_parse.netloc:
+                        sys.setrecursionlimit(1200) #FIXME: figure out what's causing a RecursionError. Python's regex module should not recurse so far that it causes this to happen.
```

Comment:
This looks sus. Check the stack trace. If this keeps being an issue then the regex that parse_qs or parse_qsl uses is too complicated and you will need to write an iterative version. You do not want people crashing the darn server from writing long links.